xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -2050,13 +1937,17 @@ bool CGUIDialogVideoInfo::ManageVideoItemArtwork(const std::shared_ptr<CFileItem
   else if (type == "actor")
     currentThumb = videodb.GetArtForItem(item->GetVideoInfoTag()->m_iDbId, item->GetVideoInfoTag()->m_type, artType);
   else
-  { // SEASON, SET
+  { // SEASON, SHOW, SET
     artType = ChooseArtType(*item);
     if (artType.empty())
       return false;
 
     if (artType == "fanart" && type != MediaTypeVideoCollection)
-      return OnGetFanart(item);
+    {
+      const bool result = OnGetFanart(item);
+      finished = false;
```

Comment:
Why are all these finished required?

Reply:
> Why are all these finished required?
 To keep existing behavior with new implementation.

User flow is open video info dialog -> select "choose art" -> open art type selection dialog, video info dialog closes -> select art type -> open art file selection dialog, art type selection dialog closes

Now there are several possibilities to end the art file selection dialog, 1. pick a file, 2. do not pick a file

In any case the art type selection dialog shall reappear. The art type selection dialog shall only disappear if you escape from it.

So, 'finished' is for indicating whether the art type selection dialog should be closed.

As I said, this is existing behavior and it took me some brain time to get the reimplemantation right.

Reply:
Callers of public `CGUIDialogVideoInfo::ManageVideoItemArtwork` want to know whether operation was successful (return value), finished is for controlling whether private `CGUIDialogVideoInfo::ManageVideoItemArtwork` has to be called again after return.

Reply:
I'd say it's fine.
Usually those bools as arguments are a bad sign but since this is only used in the context of a specific dialog and the method itself is private keep it as is